### ISSUE TITLE
FTP backups are creating empty zip files under ubuntu.

### ISSUE THREAD

**Issue by anthonyrossbach** [serghey-rodin/vesta#515]
I have no idea why but when I add a backup FTP server for my nodes they all create the zip on the server, but it is left empty, both on a fresh FTP only backup server, and also on a backup server running vestacp with a FTP account.

example backup

```
-- SYSTEM --
2015-11-19 23:40:01 68snl9p5.conf
2015-11-19 23:40:01 stats.log
2015-11-19 23:40:01 history.log
2015-11-19 23:40:01 pam

-- WEB --
2015-11-19 23:40:01 status.twii.me
2015-11-19 23:40:01 *** 1 domain ***

-- DNS --
2015-11-19 23:40:01 status.twii.me
2015-11-19 23:40:01 *** 1 domain ***

-- MAIL --
2015-11-19 23:40:01 status.twii.me
2015-11-19 23:40:01 *** 1 domain ***

-- DB --
2015-11-19 23:40:01 *** 0 databases ***

-- CRON --
2015-11-19 23:40:01 *** 0 jobs ***

-- User Dir --
2015-11-19 23:40:01 adding  .bash_logout
2015-11-19 23:40:01 adding  .bashrc
2015-11-19 23:40:01 adding  .profile
2015-11-19 23:40:01 adding  tmp
2015-11-19 23:40:01 *** 4 directories ***

-- SUMMARY --
2015-11-19 23:40:02 Local: /backup/68snl9p5.2015-11-19.tar
2015-11-19 23:42:09 Size: 1 Mb
2015-11-19 23:42:09 Runtime: 1 minute
```

**Comment by anthonyrossbach** [comment]
Was this actually fixed as I still have the issue under the latest version, I had to design a remote backup bot to do it myself.

**Comment by qSebastiaNp** [comment]
This is **still** a problem although this is ignored for > 2 years by the developers (+ there are many threads about this).
When you configure backup via FTP, Vesta's v-backup-user script opens an active FTP connection to the server. This is because it uses `ftp -n` to connect to the server, not `ftp -np` - and the UI does not let you chose whether you want passive ftp (which works out of the box, because Vesta has an allow rule for ports 21,12000-12100).
So there are two solutions:
- Stick with active FTP and add port 20 (DATA) to the allow rule in Vesta's firewall
- When the server doesn't allow active FTP or you do not want it, patch ´ftpc()´ in /usr/local/vesta/bin/v-backup-user to use ´ftp -np´

**Comment by anthonyrossbach** [comment]
I built my own system to pull and backup the backup files remote, then set the backups on origin to 0 bytes. I can also from a panel do reverse to restore, it also verifies the RAR to see if it's good and does a byte to byte check. I run a web hosting company and we now have around 30 servers running Vesta so most stuff we just started building or own ways around them.

It's all using SFTP over SSH to get in and download the backups one at a time.
